python/ray/serve/tests/test_standalone.py

```diff
@@ -925,5 +928,47 @@ def test_run_graph_task_uses_zero_cpus():
     ray.shutdown()
 
 
+@pytest.mark.parametrize(
+    "options",
+    [
+        {
+            "proxy_location": None,
+            "http_options": None,
+            "expected": HTTPOptions(location=DeploymentMode.EveryNode),
+        },
+        {
+            "proxy_location": None,
+            "http_options": HTTPOptions(location="NoServer"),
+            "expected": HTTPOptions(location=DeploymentMode.NoServer),
+        },
+        {
+            "proxy_location": None,
+            "http_options": {"location": "NoServer"},
+            "expected": HTTPOptions(location=DeploymentMode.NoServer),
+        },
+        {
+            "proxy_location": "NoServer",
+            "http_options": None,
+            "expected": HTTPOptions(location=DeploymentMode.NoServer),
+        },
+        {
+            "proxy_location": "NoServer",
+            "http_options": HTTPOptions(host="foobar"),
+            "expected": HTTPOptions(location=DeploymentMode.NoServer, host="foobar"),
+        },
+        {
+            "proxy_location": "NoServer",
+            "http_options": {"host": "foobar"},
+            "expected": HTTPOptions(location=DeploymentMode.NoServer, host="foobar"),
+        },
+    ],
+)
+def test_serve_start_proxy_location(ray_shutdown, options):
```

Comment:
Can we add test case for when both `proxy_location` and `http_options["location"]` is specified? (I'm guessing `proxy_location` takes priority?) 
Also, can we add a test case for when `http_options = {}` instead of None?

Reply:
Added both